src/emqttc_protocol.erl

```diff
@@ -119,6 +122,9 @@ init_willmsg([{retain, Retain} | Opts], WillMsg) when is_boolean(Retain) ->
 init_willmsg([_Opt | Opts], State) ->
     init_willmsg(Opts, State).
 
+refresh_password([{password_refresher, PasswordRefresher}], State) when is_function(PasswordRefresher) ->
```

Comment:
what happens when PasswordRefresher os `undefined` ?

Reply:
makes sense, added a scenario for that